internal/observability/call.go

```diff
@@ -496,6 +499,17 @@ func (c call) logStreamEvent(err error, success bool, succMsg, errMsg string, ex
 	ce.Write(fields...)
 }
 
+// emitSpanErrorTag sets the error information as tags on the current span
+func (c call) emitSpanErrorTag(res callResult) {
+	if span := opentracing.SpanFromContext(c.ctx); span != nil {
+		// emit application error to span tag if applicable
+		if res.isApplicationError {
+			transport.UpdateSpanWithoutErrMsg(span,
+				res.err, yarpcerrors.FromError(res.err).Code())
```

Comment:
Could it be that error will be set here already? In other PRs you're setting errors, including message, without checking if it's an application error.

Should we replace all other PRs with changes inside observability/call?